### PR TITLE
[test] ECC: check the bounds for auto computing cofactor

### PR DESCRIPTION
**Pull request by bbbrumley**
#9727 regression tests check that the lib can correctly compute the cofactor for all named curves. But it doesn't cover the explicit curve cases where it can't.

This regression tests illustrates the (rough) bounds stated in the `ec_guess_cofactor` comment.

As you can see, it's quite robust. Neither of these curves are remotely fit for cryptographic use, yet even with such a poor `order/cofactor` ratio the library manages until that ratio shrinks.

I don't think it's super necessary for 1.1.1 or 1.1.0.

### CODE REVIEW

**Comment by romen** [comment]
I would favor backport to 1.1.1, should cherry-pick cleanly and robustness testing is a good thing!

File: test/ectest.c [hunk 1]
```diff
@@ -1855,6 +1855,96 @@ static int parameter_test(void)
...
+ *
+ * For standards-compliant curves, cofactor is chosen as small as possible.
+ * So you can see neither of these curves are fit for cryptographic use.
+ *
+ * Some standards even mandate an upper bound on the cofactor, e.g. SECG1 v2:
+ * h <= 2**(t/8) where t is the security level of the curve, for which the lib
+ * will always succeed in computing the cofactor. Neither of these curves
+ * conform to that -- this is just robustness testing.
+ */
+static int cofactor_range_test()
```
Comment (romen) [inline test/ectest.c]: Include `void`: `static int cofactor_range_test(void)`

Travis red cross is due to this.

Comment (bbbrumley) [reply test/ectest.c]: > Travis red cross is due to this.

Thanks -- fixed in 7d2d42573758d54a2e917c6e72ba08aa6595d13b

**Comment by romen** [comment]
I will wait for the tests to end before merging.

**Comment by mattcaswell** [comment]
Looks like the tests passed.

**Comment by romen** [comment]
Merging now!

**Comment by romen** [comment]
Merged to `master` as 1d3cd983f56e0a580ee4216692ee3c9c7bf14de9
Merged to `1.1.1-stable` as 73a683b742b344072e36aab173493f5101992c30

Thanks @bbbrumley and @mattcaswell

**Pull request closed without merging**